packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.test.tsx

```diff
@@ -37,7 +42,79 @@ const localizedTexts = {
   },
 };
 describe('<AdapterDayjs />', () => {
-  describe('Localization', () => {
+  describeGregorianAdapter(AdapterDayjs, { formatDateTime: 'YYYY-MM-DD HH:mm:ss' });
+
+  describe('Adapter localization', () => {
+    describe('Russian', () => {
+      const adapter = new AdapterDayjs({ instance: dayjs, locale: 'ru' });
+      const date = adapter.date(TEST_DATE_ISO)!;
+
+      it('getWeekdays: should start from monday', () => {
+        const result = adapter.getWeekdays();
+        expect(result).to.deep.equal(['пн', 'вт', 'ср', 'чт', 'пт', 'сб', 'вс']);
+      });
+
+      it('getWeekArray: week should start from monday', () => {
+        const result = adapter.getWeekArray(date);
+        expect(result[0][0].format('dd')).to.equal('пн');
+      });
+
+      it('is12HourCycleInCurrentLocale: properly determine should use meridiem or not', () => {
+        expect(adapter.is12HourCycleInCurrentLocale()).to.equal(false);
+      });
+
+      it('getCurrentLocaleCode: returns locale code', () => {
+        expect(adapter.getCurrentLocaleCode()).to.equal('ru');
+      });
+    });
+
+    describe('English', () => {
+      const adapter = new AdapterDayjs({ instance: dayjs, locale: 'en' });
+      const date = adapter.date(TEST_DATE_ISO)!;
+
+      it('getWeekdays: should start from sunday', () => {
+        const result = adapter.getWeekdays();
+        expect(result).to.deep.equal(['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa']);
+      });
+
+      it('getWeekArray: week should start from sunday', () => {
+        const result = adapter.getWeekArray(date);
+        expect(result[0][0].format('dd')).to.equal('Su');
+      });
+
+      it('is12HourCycleInCurrentLocale: properly determine should use meridiem or not', () => {
+        expect(adapter.is12HourCycleInCurrentLocale()).to.equal(true);
+      });
+    });
+
+    it('Formatting', () => {
+      const adapter = new AdapterDayjs({ locale: 'en-US' });
```

Comment:
Nitpick, on L72 we use `locale: 'en'` and here it's `locale: 'en-US'`. Would it make sense sticking to one or the other? 🤔